build.sbt

```diff
@@ -7,8 +7,11 @@ lazy val backend = (project in file("backend"))
   .dependsOn(core % "test->test;compile->compile")
   .settings(backendSettings:_*)
 
+lazy val gcsfilesystem = (project in file("gcsfilesystem"))
+  .settings(gcsFileSystemSettings:_*)
+
 lazy val engine = (project in file("engine"))
-  .dependsOn(core % "test->test;compile->compile", backend)
+  .dependsOn(core % "test->test;compile->compile", gcsfilesystem % "test->test;compile->compile", backend)
```

Comment:
Well, if the engine does need filesystem capabilities then yes, it can depend on this, but then it would probably have to depend on every possible filesystem available.
I think if we can avoid having the engine needing filesystems it'll be better